src/main/java/seedu/us/among/ui/MainWindow.java

```diff
@@ -167,20 +172,75 @@ public EndpointListPanel getEndpointListPanel() {
         return endpointListPanel;
     }
 
+    /**
+     * Updates the progress message.
+     */
+    private void updateProgress() {
+        if (progressMessage.length() < 21) {
+            progressMessage += ".";
+        } else {
+            progressMessage = "Processing Command";
+        }
+        resultDisplay.setFeedbackToUser(progressMessage);
+    }
+
+    /**
+     * Gets the status for command execution.
+     * @return true or false to indicate if command execution is still in progress
+     */
+    public static boolean getProgress() {
+        return isInProgress;
+    }
+
+    /**
+     * Continuously update the UI to show progress to user.
+     * @return thread that is updating the UI
+     */
+    public Thread showProgress() {
+        Thread thread = new Thread(new Runnable() {
+
+            @Override
+            public void run() {
+                Runnable updater = () -> updateProgress();
+                while (MainWindow.getProgress()) {
+                    try {
+                        Thread.sleep(1000);
+                    } catch (InterruptedException ex) {
+                        //try-catch to allow thread to sleep
+                    }
+                    Platform.runLater(updater);
```

Comment:
A possible future refactor could be moving the animation to CSS and have it triggered at the start of execution and end when the result returns. But this is decent at this stage.

Reply:
I agree the animation should be further improved down the line :D